File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-games/src/services/gamesService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { GamesPageResult } from "../types/gamesModels.js";

/**
 * Games service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface GamesService {
  readonly capability: "games";
}

export function createGamesService(client: AppstoreAppClient): GamesService {
  void client;
  return { capability: "games" };
}

export function emptyGamesResult(): GamesPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_games/lib/src/services/games_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/games_models.dart';

/// Games service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class GamesService {
  const GamesService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'games';

  GamesPageResult<Never> empty() =>
      const GamesPageResult<Never>(items: <Never>[]);
}
